test/e2e/components/vl-button.js

```diff
@@ -66,18 +72,18 @@ class VlButton extends VlElement {
         return (await this.getPillType()) === 'error';
     }
 
-    async isInputAddon() {
-        return this.hasClass('vl-input-addon');
-    }
+}
 
-    async hasIcon() {
-        try {
-            await this.getIcon();
-            return true;
-        } catch(e) {
-            return false;
-        }
+class VlButtonInputAddon extends VlInputAddon {
+
+    async getIcon() {
+        return new VlIcon(this.driver, this.selector + ' [is="vl-icon"]');
```

Comment:
Dit kunnen we niet doen zonder een dependency op vl-ui-icon normaal gezien. Het is momenteel enkel een dev-dependency, dus dat wordt niet vanzelf binnengetrokken bij de afnemers, waardoor het component object niet bruikbaar zal zijn.

Reply:
Besproken met @SanderKleykens. Best de require verplaatsen naar de `getIcon` functie en de VlIcon maar aanmaken indien er een element met vl-icon is attribuut bestaat.